src/anemoi/graphs/plotting/interactive_html.py

```diff
@@ -0,0 +1,234 @@
+from typing import Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import plotly.graph_objects as go
+from torch_geometric.data import HeteroData
+
+from anemoi.graphs.plotting.prepare import compute_isolated_nodes
+from anemoi.graphs.plotting.prepare import compute_node_adjacencies
+from anemoi.graphs.plotting.prepare import edge_list
+from anemoi.graphs.plotting.prepare import node_list
+
+annotations_style = {"text": "", "showarrow": False, "xref": "paper", "yref": "paper", "x": 0.005, "y": -0.002}
+plotly_axis_config = {"showgrid": False, "zeroline": False, "showticklabels": False}
+
+LOGGER = logging.getLogger(__name__)
+
+
+def plot_interactive_subgraph(
+    graph: HeteroData,
+    edges_to_plot: tuple[str, str, str],
+    out_file: Optional[Union[str, Path]] = None,
+) -> None:
+    """Plots a bipartite graph (bi-graph).
+
+    This methods plots the bipartite graph passed in an interactive window (using Ploty).
+
+    Parameters
+    ----------
+    graph : dict
+        The graph to plot.
+    edges_to_plot : tuple[str, str]
+        Names of the edges to plot.
+    out_file : str | Path, optional
+        Name of the file to save the plot. Default is None.
+    """
+    source_name, _, target_name = edges_to_plot
+    edge_x, edge_y = edge_list(graph, source_nodes_name=source_name, target_nodes_name=target_name)
+    assert source_name in graph.node_types, f"edges_to_plot ({source_name}) should be in the graph"
+    assert target_name in graph.node_types, f"edges_to_plot ({target_name}) should be in the graph"
+    lats_source_nodes, lons_source_nodes = node_list(graph, source_name)
+    lats_target_nodes, lons_target_nodes = node_list(graph, target_name)
+    dst_num_nodes = graph[target_name].num_nodes
+    node_adjacencies, node_text = compute_node_adjacencies(graph, source_name, target_name, dst_num_nodes)
+
+    edge_trace = go.Scattergeo(
+        lat=edge_x,
+        lon=edge_y,
+        line={"width": 0.5, "color": "#888"},
+        hoverinfo="none",
+        mode="lines",
+        name="Connections",
+    )
+
+    source_node_trace = go.Scattergeo(
+        lat=lats_source_nodes,
+        lon=lons_source_nodes,
+        mode="markers",
+        hoverinfo="text",
+        name=source_name,
+        marker={
+            "showscale": False,
+            "color": "red",
+            "size": 2,
+            "line_width": 2,
+        },
+    )
+
+    target_node_trace = go.Scattergeo(
+        lat=lats_target_nodes,
+        lon=lons_target_nodes,
+        mode="markers",
+        hoverinfo="text",
+        name=target_name,
+        text=node_text,
+        marker={
+            "showscale": True,
+            "colorscale": "YlGnBu",
+            "reversescale": True,
+            "color": node_adjacencies,
+            "size": 10,
+            "colorbar": {"thickness": 15, "title": "Node Connections", "xanchor": "left", "titleside": "right"},
+            "line_width": 2,
+        },
+    )
+    layout = go.Layout(
```

Comment:
Happy to discuss this. But I think in the future it could be worth refactoring this script in a similar fashion as we have in aifs-mono the callbacks.py/plots.py, with functions like _init_plot_settings or base_plot class where we can abstract the common functionality between all the plotting functions and then simply have each function to define the specifics of that method